### ISSUE TITLE
Object of class GMP could not be converted to boolean

### ISSUE THREAD

**Issue by jchook** [bobthecow/psysh#557]
Similar to #433, psysh seems to want to convert a [GMP](https://www.php.net/manual/en/class.gmp.php) object to boolean for inspection.

![image](https://user-images.githubusercontent.com/193530/55690644-28f01300-5962-11e9-8ff1-dceefbb3bde6.png)

This does not happen with `php -a` for example:

![image](https://user-images.githubusercontent.com/193530/55690663-7e2c2480-5962-11e9-8982-fb2fe633362c.png)

Seems to happen any time PHP tries to cast a GMP into a boolean, even internally, e.g. `if ($x)`

![image](https://user-images.githubusercontent.com/193530/55690680-b6336780-5962-11e9-9ed7-8a26793057a6.png)

**Comment by bobthecow** [comment]
If there's a problem dumping GMP for inspection, it'd be in Symfony VarDumper, but I'm not seeing that with the latest VarDumper:

```
Psy Shell v0.9.9 (PHP 7.3.3 — cli) by Justin Hileman
>>> $x = gmp_mul(2,12)
=> GMP {#2330
     value: 24,
   }
```

The error when implicitly casting to boolean for `if` happens in `php -a` for me, so that seems like psysh is doing the right thing:

```
$ php -a
Interactive shell

php > $x = gmp_mul(2,12);
php > if ($x) echo "x";
PHP Recoverable fatal error:  Object of class GMP could not be converted to bool in php shell code on line 1

Recoverable fatal error: Object of class GMP could not be converted to bool in php shell code on line 1
```

**Comment by jchook** [comment]
Interesting, so VarDumper already fixed the issue.

Will that go out with the next release? Do I have outdated software on my end?

**Comment by bobthecow** [comment]
I just checked, and PHP 7.3 alone doesn't fix it, but the fix in VarDumper might only be available on 7.3?

installing via composer is the right call. the version of psysh you can get via wget is somewhat of a compatibility build, so it won't necessarily have the latest and greatest dependency versions.

maybe try switching to the develop branch for your composer install, and see whether that gets things working in 7.2?

**Comment by bobthecow** [comment]
Closing as this was an upstream issue and appears to be resolved (a while ago).
